topological_navigation/scripts/localisation.py

```diff
@@ -161,6 +163,27 @@ def Callback(self, msg, item):
 
 
 
+    def get_nodes_wtag_cb(self,req):
+        tlist = []
+        rlist=[]
+
+        rospy.wait_for_service('/topological_map_manager/get_tagged_nodes')
+        try:
+            cont = rospy.ServiceProxy('/topological_map_manager/get_tagged_nodes', strands_navigation_msgs.srv.GetTaggedNodes)
+            resp1 = cont(req.tag)
+            tagnodes = resp1.nodes
+        except rospy.ServiceException, e:
+            print "Service call failed: %s"%e
+            rlist.append(tlist)
+            return rlist
```

Comment:
why still returning anything? If this failed then something's gone massive wrong, right? Now the caller what just assumes there wasn't anything by their tag name. But that's not true.